pkg/kvstore/store/syncstore.go

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package store
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"path"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+// SyncStore abstracts the operations allowing to synchronize key/value pairs
+// into a kvstore.
+type SyncStore interface {
+	// Run starts the SyncStore logic, blocking until the context is closed.
+	Run(ctx context.Context)
+
+	// UpsertKey upserts a key/value pair into the kvstore.
+	UpsertKey(ctx context.Context, key Key) error
+
+	// DeleteLocalKey removes a key from the kvstore.
+	DeleteKey(ctx context.Context, key NamedKey) error
```

Comment:
Concerning the interface definition, I'm wondering whether to keep the context as parameter for the `UpsertKey` and `DeleteKey` methods. The current workqueue-based implementation ignores it, but it might make sense in a possible synchronous implementations.